Movement.cpp

```diff
@@ -6,154 +6,75 @@
 */
 #include "Movement.h"
 
-Movement::Movement() {}
+Movement::Movement(){}
 
-void Movement::advanceP(const double desire) {
-  encouder_count_left = 0;
-  encouder_count_right = 0;
+Movement::Movement(BNO *b, Control *s, Motors *r) {
+    bno_ = b;
+    system_ = s;
+    robot_ = r;
+}
+
+void Movement::advancePID(const double desire) {
+  encoder_count_left = 0;
+  encoder_count_right = 0;
 
   do {
-      double pwm_right_enginees = bno_.getPwmBNORight(desire); // Negative.
-      double pwm_left_enginees = bno_.getPwmBNOLeft(desire); // Positive. 
+      double pwm_right_enginees = system_->getPwmBNORight(desire); // Negative.
+      double pwm_left_enginees = system_->getPwmBNOLeft(desire); // Positive. 
       if (pwm_left_enginees > 1) {
-          pwm_left_enginees = kLimit_inf_pwm;
-          control.getPwm(pwm_left_enginees);
+          pwm_right_enginees = Common::kLimit_inf_pwm;
+          system_->getPwm(pwm_left_enginees);
       }
```

Comment:
Do you still need to do this check or do the functions getPwmBNORight and Left already do it?
Also, on getPWMBNORight() you shouldn't call the Ultrasonic one. They should be two independent functions. What if I don't want to use the Ultrasonics or the other way around?

Reply:
On the functions getPwnoBNORight and Left, I do not call the "getPwm", the same with the PwmUltrasonicRight and Left, I call "getPwm" after I put all the Pwm together.